lms/djangoapps/discussion_api/tests/test_api.py

```diff
@@ -515,4 +605,43 @@ def test_pagination(self):
         # Test page past the last one
         self.register_get_threads_response([], page=3, num_pages=3)
         with self.assertRaises(Http404):
-            get_thread_list(self.request, self.course.id, page=4, page_size=10)
+            get_thread_list(self.request, self.course, page=4, page_size=10)
+
+    @ddt.data(
+        (FORUM_ROLE_ADMINISTRATOR, True, False, True),
+        (FORUM_ROLE_ADMINISTRATOR, False, True, False),
+        (FORUM_ROLE_MODERATOR, True, False, True),
+        (FORUM_ROLE_MODERATOR, False, True, False),
+        (FORUM_ROLE_COMMUNITY_TA, True, False, True),
+        (FORUM_ROLE_COMMUNITY_TA, False, True, False),
+        (FORUM_ROLE_STUDENT, True, False, True),
+        (FORUM_ROLE_STUDENT, False, True, True),
+    )
+    @ddt.unpack
+    def test_anonymity(self, role_name, thread_anon, thread_anon_to_peers, expected_anon):
```

Comment:
thread_anon is the value of the thread's "anonymous" field
thread_anon_to_peers is the value of the thread's "anonymous_to_peers" field
expected_anon is whether the thread should actually be anonymous to a user with the given role